### PR TITLE
Use double quotes for variables in IF

### PR DESCRIPTION
**Pull request by rninov**
Use double quotes for variables in if and use `-eq` for compare numbers

#### Proposed Changes ####

Make more more secure by surrounding variable with double quotes
#### Types of Changes ####

Bugfix
#### User-Facing Change ####

```release-note
NONE
```

### CODE REVIEW

**Comment by briandowns** [comment]
We require all all commits be signed:

```sh
git commit --amend --no-edit --signoff
git push --force-with-lease origin patch-1
```

File: scripts/binary_size_check.sh [hunk 1]
```diff
@@ -13,16 +13,16 @@ fi
...
-elif [ ${ARCH} = arm ]; then
+elif [ "${ARCH}" = arm ]; then
     BIN_SUFFIX="-armhf"
 fi
 
 CMD_NAME="dist/artifacts/k3s${BIN_SUFFIX}"
 SIZE=$(stat -c '%s' ${CMD_NAME})
 
-if [ ${SIZE} -gt ${MAX_BINARY_SIZE} ]; then
+if [ "${SIZE}" -gt |${MAX_BINARY_SIZE}| ]; then
```
Comment (briandowns) [inline scripts/binary_size_check.sh:25]: What is the intent of the pipes surrounding the variable here?

File: scripts/binary_size_check.sh [hunk 2]
```diff
@@ -2,7 +2,7 @@
 
 set -e
 
-if [ "${DEBUG}" = 1 ]; then
+if [ "${DEBUG}" -eq 1 ]; then
```
Comment (briandowns) [inline scripts/binary_size_check.sh:5]: What benefit are we gaining from this change?

Comment (rninov) [reply scripts/binary_size_check.sh:5]: We compare numbers according to the rules in bash.

File: scripts/binary_size_check.sh (hunk 1 above)
Comment (rninov) [reply scripts/binary_size_check.sh:25]: This is my mistake, is must be double quotes.

File: scripts/binary_size_check.sh (hunk 2 above)
Comment (Oats87) [reply scripts/binary_size_check.sh:5]: @rninov for future reference, our goal is to have the K3s scripts POSIX compliant, so we do not allow bashisms/bash-specific implementations of functionality.

For this PR and this specific change, I think this is sane as it is implementing a change from string comparison to algebraic integer comparison.

**Review by Oats87 — APPROVED** [review]

**Comment by Oats87** [comment]
@rninov As @briandowns has said, you need to make sure your commits are signed off or else we cannot merge your PR.

**Review by briandowns — APPROVED** [review]

**Comment by dereknola** [comment]
@rninov Please signoff on your commits, or this PR will be closed.

**Comment by rninov** [comment]
> @rninov Please signoff on your commits, or this PR will be closed.

Feel free to close it

**Pull request closed without merging**